pkgs/development/libraries/zlib/default.nix

```diff
@@ -101,9 +101,15 @@ stdenv.mkDerivation (finalAttrs: {
     ln -s zlib1.dll $out/bin/libz.dll
   '';
 
-  # As zlib takes part in the stdenv building, we don't want references
-  # to the bootstrap-tools libgcc (as uses to happen on arm/mips)
-  env.NIX_CFLAGS_COMPILE = lib.optionalString (!stdenv.hostPlatform.isDarwin) "-static-libgcc";
+  env = lib.optionalAttrs (!stdenv.hostPlatform.isDarwin) {
+    # As zlib takes part in the stdenv building, we don't want references
+    # to the bootstrap-tools libgcc (as uses to happen on arm/mips)
+    NIX_CFLAGS_COMPILE = "-static-libgcc";
+  } // lib.optionalAttrs (stdenv.hostPlatform.linker == "lld") {
+    # lld 16 enables --no-undefined-version by defualt
+    # This makes configure think it can't build dynamic libraries
+    NIX_LDFLAGS = "--undefined-version";
```

Comment:
I filed an upstream issue about this! Maybe we should link to it. https://github.com/madler/zlib/issues/960

Reply:
force pushed to add a comment with that link! thanks for filing :)